Refactor/sky rev fwd

ersonp opened the pull request:
Did you run `make format && make check`?

Fixes #	

 Changes:	
-	

How to test this PR:

Commit:
refactor: Rename `skyrev` and `skyfwd` to `net`

Rename the `skyrev` package to `con` and `skyfwd` to `pub` under the `net` command. This change improves the clarity and consistency of the codebase.

Commit:
refactor: Rename `forward` to `connect`

Commit:
feat: Add `publish` and update `connect`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/skycoin/skywire?pullRequest=1830) <br/>All committers have signed the CLA.

ersonp committed:
fix(publish): Remove ReadTimeout

ersonp committed:
fix(connect): Add sync.Mutex to handleConnectFunc

ersonp committed:
refactor: Remove `RegisterHTTPPort` and `DeregisterHTTPPort` from API and RPC

ersonp committed:
chore(visor/api): Fix go-staticcheck warnings

ersonp committed:
chore: Update .gitignore to ignore skywire-config.json and local directories

ersonp committed:
refactor: Convert remaning skyfwd code to `net`

ersonp committed:
refactor: Rename and change `ListHTTPPorts` and `ListConnected`

ersonp committed:
refactor: Update ConnectConn to use http.Server for serving

ersonp committed:
chore: Update comment

ersonp committed:
refactor: Update `connect` command to print connection details in JSON format

ersonp committed:
feat: add app type support

Add support for HTTP, TCP, UDP in the API, RPC and CMD

ersonp committed:
feat: ensure port availability

 Ensure port availability before adding ConnectConn and PublishListener

ersonp committed:
refactor: update APIs

Update `Connect` and `Publish` API to return ConnectInfo and PublishInfo as well as for `ListPublished` and `ListConnected` to return []PublishInfo and []ConnectInfo

ersonp committed:
refactor: add subcommands to con and pub

ersonp committed:
fix(publish): remove http.Server headers

Remove ReadHeaderTimeout and WriteTimeout that were causing the publisher to close the incoming connection if nothing was connectead to it withing 5 seconds.

ersonp committed:
chore: fix linting issues

ersonp committed:
chore: fix linting

ersonp committed:
fix(publish): impliment proper usage of LocalPort

ersonp committed:
refactor: Update `connect` and `publish` commands

ersonp committed:
refactor: Update `publish.go` to address Slowloris attack vector

ersonp committed:
chore: fix linting

ersonp committed:
chore: fix linting

0pcom commented:
Closing this PR to clean the repo ; i will finish re-implementing this in the future

please delete your fork
